s3/src/main/java/com/instaclustr/kafka/connect/s3/AwsStorageConnectorCommonConfig.java

```diff
@@ -61,7 +64,7 @@ public static void verifyS3CredentialsAndBucketInfo(final Map<String, String> se
             String awsRegion = sentConfigMap.get(AWS_REGION);
             AmazonS3 s3Client = TransferManagerProvider.getS3ClientBuilderWithRegionAndCredentials(sentConfigMap).build();
             if (s3Client.doesBucketExistV2(s3BucketName)) {
-                if (awsRegion != null) {
+                if (StringUtils.isBlank(sentConfigMap.get(S3_ENDPOINT)) && awsRegion != null) {
```

Comment:
So in case of endpoint, the region doesn't matter?

Reply:
Yes that is right. 
For non-aws endpoints this code throws an exception,

>  Region.fromValue(s3Client.getBucketLocation(s3BucketName)).toAWSRegion().getName();

I believe it's because the client does not contain information on the region when it's connecting to a non-aws S3 buckets.